.github/workflows/GHA-Scala-Functional-Tests.yaml

```diff
@@ -72,9 +72,10 @@ jobs:
 
 
       - name: Upload coverage to Codecov
-        if: success()
+        if: matrix.java-version == '11'
```

Comment:
I've been reading the docs and they are confusing me a bit.
It says if any previous step fails, success() will return false. So, if a first run of unit test fails and we continue on error, and unit test pass on the first or second retry, that value should still return false and we will not upload reports.
But, maybe I misunderstand something and it will reset? 
I can test if you think we should.

Reply:
Actually, looking at the debug in the output it shows. I think this is just default then.
`##[debug]Evaluating: (success() && (matrix.java-version == '17'))`

Reply:
```
##[debug]Evaluating condition for step: 'Upload coverage to Codecov'
##[debug]Evaluating: (success() && (matrix.java-version == '17'))
##[debug]Evaluating And:
##[debug]..Evaluating success:
##[debug]..=> true
```

Reply:
I decided to include a yml file here for the expired uploads.
The reason we hit this issue is if we re-run reports for one but not all tests and it has been longer than 12hrs, tests that do not rerun will cause error on the upload and limits the reporting in codecov for those older report files.
https://docs.codecov.com/docs/codecov-yaml#expired-reports
This was suggested to us here based on the issue I reported to codecov:
https://community.codecov.com/t/github-action-upload-success-codecov-upload-expired/4332